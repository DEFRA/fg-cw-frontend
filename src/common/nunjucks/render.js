import camelCase from "lodash/camelCase.js";
import { environment } from "./nunjucks.js";

/**
 * @param {string} name
 * @param {object} params
 * @param {string} [block]
 */
export const render = (name, params, block) => {
  const macro = {
    path: `${name}/macro.njk`,
    name: camelCase(name.split("-")),
    params: JSON.stringify(params, null, 2),
  };

  const callComponent = block
    ? `{%- call ${macro.name}(${macro.params}) -%}${block}{%- endcall -%}`
    : `{{- ${macro.name}(${macro.params}) -}}`;

  return environment.renderString(
    `{%- from "${macro.path}" import ${macro.name} -%}
    ${callComponent}
    `,
    {},
  );
};
