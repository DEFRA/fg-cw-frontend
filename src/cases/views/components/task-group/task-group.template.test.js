import { describe, expect, test } from "vitest";
import { render } from "../../../../common/nunjucks/render.js";

describe("task-group", () => {
  test("renders", () => {
    const component = render("task-group", {
      caseId: "123",
      stage: {
        name: "Test Stage",
        taskGroups: [
          {
            name: "Group 1",
            tasks: [
              {
                name: "Task 1",
                status: "not started",
                link: "/cases/123/task/1",
              },
              {
                name: "Task 2",
                status: "in progress",
                link: "/cases/123/task/2",
              },
            ],
          },
        ],
        actions: [
          {
            label: "Complete Stage",
            nextStage: "complete",
          },
        ],
      },
    });

    expect(component).toMatchSnapshot();
  });
});
