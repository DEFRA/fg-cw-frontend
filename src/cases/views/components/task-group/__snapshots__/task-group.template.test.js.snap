// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`task-group > renders 1`] = `
"
<div class="govuk-grid-row">
  <div class="govuk-grid-column-two-thirds">
    <h2
      class="govuk-heading-l govuk-!-margin-top-6"
      data-testid="stage-heading"
    >
      Test Stage
    </h2>
      <h3 class="govuk-heading-m">1. Group 1</h3>
      
  <ul
    class="govuk-task-list govuk-!-margin-bottom-4"
    data-testid="taskList-list"
  >
      <li
        class="govuk-task-list__item govuk-task-list__item--with-link"
        data-testid="taskList-li"
      >
        <div class="govuk-task-list__name-and-hint">
          <a href="/cases/123/task/1" class="govuk-link govuk-task-list__link"
            >Task 1</a
          >
        </div>
        <div class="govuk-task-list__status">
          <strong class="govuk-tag govuk-tag govuk-tag--blue">
  Not started
</strong>

        </div>
      </li>
      <li
        class="govuk-task-list__item govuk-task-list__item--with-link"
        data-testid="taskList-li"
      >
        <div class="govuk-task-list__name-and-hint">
          <a href="/cases/123/task/2" class="govuk-link govuk-task-list__link"
            >Task 2</a
          >
        </div>
        <div class="govuk-task-list__status">
          <strong class="govuk-tag govuk-tag govuk-tag--blue">
  In progress
</strong>

        </div>
      </li>
  </ul>


      <div class="govuk-button-group">
    <form method="POST">
      <button
        type="submit"
        data-prevent-double-click="true"
        class="govuk-button"
        data-module="govuk-button"
        data-govuk-button-init=""
        formaction="/cases/123"
disabled      >
        Complete Stage
      </button>
    </form>
</div>

  </div>
</div>
"
`;
